skywalking/plugins/sw_urllib_request.py

```diff
@@ -38,15 +39,23 @@ def _sw_open(this: OpenerDirector, fullurl, data, timeout):
 
         context = get_context()
         carrier = Carrier()
-        with context.new_exit_span(op=fullurl.selector or '/', peer=fullurl.host, carrier=carrier) as span:
+        url = fullurl.selector.split("?")[0] if fullurl.selector else '/'
+        with context.new_exit_span(op=url, peer=fullurl.host, carrier=carrier) as span:
             span.layer = Layer.Http
             span.component = Component.General
 
             [fullurl.add_header(item.key, item.val) for item in carrier]
 
-            res = _open(this, fullurl, data, timeout)
-            span.tag(Tag(key=tags.HttpMethod, val=fullurl.get_method()))
-            span.tag(Tag(key=tags.HttpUrl, val=fullurl.full_url))
+            try:
+                res = _open(this, fullurl, data, timeout)
+                span.tag(Tag(key=tags.HttpMethod, val=fullurl.get_method()))
+            except HTTPError as e:
+                span.tag(Tag(key=tags.HttpStatus, val=e.code))
+                raise
+            finally:  # we do this here because it may change in _open()
+                span.tag(Tag(key=tags.HttpMethod, val=fullurl.get_method()))
```

Comment:
Please note that tag with the same key can co-exist, here you tag the span twice with the same key (`tags.HttpMethod`), which breaks the tests

Reply:
You are right I forgot to remove that, but I don't think that is the source of the problem, lets see...

Reply:
> You are right I forgot to remove that, but I don't think that is the source of the problem (or at least the only problem), lets see...

What potential problem do you foresee? From the code level, I think it looks good. As the test case is simple (endpoint names without `?`), so the changes should be OK IMO

Reply:
Well, because this does change behavior. If there are tests that previously did not expect these tags due to testing a fail case and they are now returned then that will be a difference.

Reply:
> Well, because this does change behavior. If there are tests that previously did not expect these tags due to testing a fail case and they are now returned then that will be a difference.

OK, that makes sense, IIRC, we didn't test fail case, so the tests should pass